nncf/extensions/__init__.py

```diff
@@ -1,5 +1,41 @@
+import enum
 import torch
 
+import os
+
+from nncf.common.utils.registry import Registry
+
+EXTENSIONS = Registry('extensions')
+
+
+class ExtensionsType(enum.Enum):
+    CPU = 0
+    CUDA = 1
+
+
+def _force_build_extensions(ext_type):
+    for class_type in EXTENSIONS.registry_dict.values():
+        if class_type.extension_type() != ext_type:
+            continue
+        class_type.load()
```

Comment:
How do you even know that `class_type` will have a `.load()` method?

Reply:
Now using an abstract class; Does this resolve an issue?